portals/publisher/source/src/app/components/Apis/Details/LifeCycle/Components/PublishWithoutDeploy.jsx

```diff
@@ -82,9 +84,109 @@ const DialogActions = withStyles((theme) => ({
  */
 export default function PublishWithoutDeploy(props) {
     const {
-        apiID, handleClick, open, handleClose,
+        classes, api, handleClick, open, handleClose,
     } = props;
 
+    let isExternalEndpointAvailable = false;
+    let availableExternalEndpoint = '';
+    if (api.advertiseInfo && api.advertiseInfo.apiExternalProductionEndpoint
+        && api.advertiseInfo.apiExternalProductionEndpoint.length > 0) {
+        isExternalEndpointAvailable = true;
+        availableExternalEndpoint = api.advertiseInfo.apiExternalProductionEndpoint;
+    } else if (api.advertiseInfo && api.advertiseInfo.apiExternalSandboxEndpoint
+        && api.advertiseInfo.apiExternalSandboxEndpoint.length > 0) {
+        isExternalEndpointAvailable = true;
+        availableExternalEndpoint = api.advertiseInfo.apiExternalSandboxEndpoint;
+    }
+
+    const [externalEndpoint, setExternalEndpoint] = useState(availableExternalEndpoint);
+    const [isValidExternalEndpoint, setValidExternalEndpoint] = useState(isExternalEndpointAvailable);
+
+    /**
+     * Validate external endpoint
+     *
+     * @param event
+     */
+    const handleOnChangeExternalEndpoint = (event) => {
+        const { value } = event.target;
+        setExternalEndpoint(value);
+        if (value && value.length > 0) {
+            let url;
+            try {
+                url = new URL(value);
+            } catch (_) {
+                setValidExternalEndpoint(false);
+            }
+            if (url) {
+                setValidExternalEndpoint(true);
+            } else {
+                setValidExternalEndpoint(false);
+            }
+        } else {
+            setValidExternalEndpoint(false);
+        }
+    };
+
+    const handlePublishClick = () => {
+        if (externalEndpoint && externalEndpoint.length > 0) {
+            const updatedAPI = {
+                id: api.id,
```

Comment:
update only the advertiseInfo property of the original API object

Reply:
this is not possible since, the given API object is not compatible for the expected REST api request body